compute_endpoint/tests/unit/test_endpoint_unit.py

```diff
@@ -158,178 +109,133 @@ def _wrapped_umask(new_umask: int | None) -> int:
-        headers={"Content-Type": "application/json"},
-        json={},
-    )
+    ep, ep_dir, *_, ep_conf = mock_ep_data
 
-    mock_gcc = get_standard_compute_client()
-    mock_delete = mocker.patch.object(
-        mock_gcc, "delete_endpoint", side_effect=NetworkError("foo", Exception)
-    )
-    mocker.patch(f"{_mock_base}Endpoint.get_funcx_client").return_value = mock_gcc
-    mocker.patch(f"{_mock_base}Endpoint.get_endpoint_id").return_value = ep_uuid
-    mock_log = mocker.patch(f"{_mock_base}log")
+    mock_gcc.delete_endpoint.side_effect = NetworkError("foo", Exception)
 
     f = io.StringIO()
     with redirect_stdout(f):
         with pytest.raises(SystemExit) as pytest_exc:
             ep.delete_endpoint(ep_dir, ep_conf, ep_uuid=ep_uuid)
 
     assert pytest_exc.value.code == os.EX_TEMPFAIL
-    assert mock_delete.called, "Verify test setup; was kernel actually invoked?"
+    assert mock_gcc.delete_endpoint.called, "Verify test: was kernel invoked?"
     assert "could not be deleted from the web" in mock_log.warning.call_args[0][0]
     assert "unable to reach the Globus Compute" in mock_log.critical.call_args[0][0]
     assert "unable to reach the Globus Compute" in f.getvalue()  # stdout
 
 
-@responses.activate
 def test_register_endpoint_invalid_response(
-    mocker,
-    fs,
-    endpoint_uuid,
-    other_endpoint_id,
-    register_endpoint_response,
-    get_standard_compute_client,
-    mock_ep_data,
+    mock_log, ep_uuid, other_endpoint_id, mock_gcc, mock_get_client, mock_ep_data
 ):
-    mock_gcc = get_standard_compute_client()
-    mock_log = mocker.patch(f"{_mock_base}log")
-    mocker.patch(f"{_mock_base}Endpoint.get_funcx_client").return_value = mock_gcc
-
     ep, ep_dir, log_to_console, no_color, ep_conf = mock_ep_data
+    ep_args = (ep_dir, ep_uuid, ep_conf, log_to_console, no_color)
 
-    responses.add(
-        method=responses.PUT,
-        url=f"https://compute.api.globus.org/v3/endpoints/{endpoint_uuid}",
-        headers={"Content-Type": "application/json"},
-        json={"endpoint_id": other_endpoint_id},
-    )
+    mock_gcc.register_endpoint.return_value = {"endpoint_id": other_endpoint_id}
 
     with pytest.raises(SystemExit) as pytest_exc:
-        ep.start_endpoint(
-            ep_dir,
-            endpoint_uuid,
-            ep_conf,
-            log_to_console,
-            no_color,
-            reg_info={},
-            ep_info={},
-        )
+        ep.start_endpoint(*ep_args, reg_info={}, ep_info={})
+
     assert pytest_exc.value.code == os.EX_SOFTWARE
-    assert "mismatched endpoint id" in mock_log.error.call_args[0][0]
-    assert "Expected" in mock_log.error.call_args[0][0]
-    assert "received" in mock_log.error.call_args[0][0]
-    assert endpoint_uuid in mock_log.error.call_args[0][0]
-    assert other_endpoint_id in mock_log.error.call_args[0][0]
+    a, _k = mock_log.error.call_args
+    assert "mismatched endpoint id" in a[0]
```

Comment:
A lot of repeats I would do something shorter like
```
assert s in mock_log.error.call_args[0][0] for s in [
    "mismatched endpoint id", "Expected", "received", ep_uuid, other_endpoint_id
]
```
But that's a nitpick, you can ignore

Reply:
Hah!  I was just happy to clean up the `.call_args[0]` to the simpler to read `a`.  But fair enough.